clang/lib/AST/DeclPrinter.cpp

```diff
@@ -1190,7 +1190,11 @@ void DeclPrinter::printTemplateParameters(const TemplateParameterList *Params,
       VisitNonTypeTemplateParmDecl(NTTP);
     } else if (auto TTPD = dyn_cast<TemplateTemplateParmDecl>(Param)) {
       VisitTemplateDecl(TTPD);
-      // FIXME: print the default argument, if present.
+      if (TTPD->hasDefaultArgument() && !TTPD->defaultArgumentWasInherited()) {
+        Out << " = ";
+        TTPD->getDefaultArgument().getArgument().print(Policy, Out,
+                                                       /*IncludeType=*/false);
+      }
```

Comment:
Default argument handling is similar for all the three kinds. Maybe, it should be a template method `VisitTemplateDefArg`? Or it is not justified here?

Moreover, maybe, `VisitTemplateDecl` call and the default argument handling should be united into a `VisitTemplateTemplateParmDecl` to be similar with the previous two branches?

Reply:
Yeah the other cases are in pretty much the same situation, and they are also only used here.

Either moving all of them inline, or moving this one to a likewise separate function seems fine to me.